cgsmiles/drawing_utils.py

```diff
@@ -104,48 +99,112 @@ def make_graph_edges(graph, pos):
     nodes have the same fragid attribute).
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        graph of the molecule with 'fragid' attribute
+    plain_edges: dict[frozenset([abc.hashable])]
+        dict of edge coordintes indexed by node keys
+
+    Returns
+    -------
+    dict[list]
+        dict of edges belonging to a paticular fragid
     """
     mapped_edges = defaultdict(list)
     for idx, jdx in graph.edges:
-        if graph.nodes[idx]['fragid'] == graph.nodes[jdx]['fragid']:
-            fragid = graph.nodes[idx]['fragid'][0]
-            mapped_edges[fragid].extend(plain_edges[frozenset([idx, jdx])])
-        elif len(graph.nodes[idx]['fragid']) > 1 and\
-             graph.nodes[jdx]['fragid'][0] in graph.nodes[idx]['fragid']:
-            fragid = graph.nodes[jdx]['fragid'][0]
+        frag_idx = graph.nodes[idx]['fragid']
+        frag_jdx = graph.nodes[jdx]['fragid']
+        # edge belongs to the same fragment
+        # e.g. [0] == [0]
+        if frag_idx == frag_jdx:
+            frag_id = frag_idx[0]
+            mapped_edges[frag_id].extend(plain_edges[frozenset([idx, jdx])])
+        # edge on idx belongs to multiple fragments and at least one
+        # fragment id is the same as in jdx e.g. [1,2] in [2] or
+        # [1, 2] & [2, 3]
+        elif len(frag_idx) > 1 and any(fid in frag_idx for fid in frag_idx):
+            fragid = frag_jdx[0]
             mapped_edges[fragid].extend(plain_edges[frozenset([idx, jdx])])
-        elif len(graph.nodes[jdx]['fragid']) > 1 and\
-             graph.nodes[idx]['fragid'][0] in graph.nodes[jdx]['fragid']:
+        # same as above
+        elif len(frag_jdx) > 1 and any(fid in frag_idx for fid in frag_idx):
             fragid = graph.nodes[idx]['fragid'][0]
             mapped_edges[fragid].extend(plain_edges[frozenset([idx, jdx])])
     return mapped_edges
 
-def make_node_pies(graph, pos, cgmapping, colors, outline=False, radius=0.2, linewidth=0.2, use_weights=False):
+def make_node_pies(graph,
+                   pos,
+                   cgmapping,
+                   colors,
+                   outline=False,
+                   radius=0.2,
+                   linewidth=0.2,
+                   use_weights=False):
     """
     Generate the slices for the matplotlip pies used to draw nodes.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        graph of the molecule
+    pos: dict[np.array]
+        dict of 2D node positions
+    cgmapping: bool
+        if the drawing includes a cgmapping;
+        uses the fragid attribute for colors
+        otherwise node keys are used
+    colors: dict
+        dict of colors
+    radius: float
+        radius fo the pie
+    linewidth: float
+        outline width of the pie
+    use_weights: bool
+        use the weight attribute when drawing node pies
+
+    Yields
+    ------
+    np.array, dict
+        array slices and keyword arguments to be given
+        to mpl.Pie class
```

Comment:
Yes, but there is a scenario where you will get an outline even if you did not ask for it. For example, if a weight is zero and the node belongs to more than one edge. Otherwise the drawing of the mapping with the alpha compoente gets messed up. So basically setting linewidth independenly from outline makes sense.